### ISSUE TITLE
PostgreSQL 16 support

### ISSUE THREAD

**Issue by devrimgunduz** [cybertec-postgresql/pg_squeeze#61]
Hi,

1.5.2 fails to build against PostgreSQL 16. Errors are below. Can you please take a look? v16 is right around the corner.. 

Thanks! Devrim

> /usr/lib64/ccache/clang -Wno-ignored-attributes -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Xclang -no-opaque-pointers -Wno-unused-command-line-argument -Wno-compound-token-split-by-macro -O2  -I. -I./ -I/usr/pgsql-16/include/server -I/usr/pgsql-16/include/internal  -D_GNU_SOURCE -I/usr/include/libxml2  -I/usr/include -flto=thin -emit-llvm -c -o pgstatapprox.bc pgstatapprox.c
pg_squeeze.c: In function 'create_transient_table':
pg_squeeze.c:2528:29: warning: implicit declaration of function 'pg_tablespace_aclcheck' [-Wimplicit-function-declaration]
 2528 |                 aclresult = pg_tablespace_aclcheck(tablespace, relowner, ACL_CREATE);
      |                             ^~~~~~~~~~~~~~~~~~~~~~
concurrent.c: In function 'apply_concurrent_changes':
concurrent.c:297:35: error: too few arguments to function 'ExecInsertIndexTuples'
  297 |                         recheck = ExecInsertIndexTuples(
      |                                   ^~~~~~~~~~~~~~~~~~~~~
In file included from concurrent.c:18:
/usr/pgsql-16/include/server/executor/executor.h:630:14: note: declared here
  630 | extern List *ExecInsertIndexTuples(ResultRelInfo *resultRelInfo,
      |              ^~~~~~~~~~~~~~~~~~~~~
concurrent.c:392:33: error: too few arguments to function 'simple_heap_update'
  392 |                                 simple_heap_update(relation, &ctid, tup);
      |                                 ^~~~~~~~~~~~~~~~~~
In file included from pg_squeeze.h:18,
                 from concurrent.c:13:
/usr/pgsql-16/include/server/access/heapam.h:278:13: note: declared here
  278 | extern void simple_heap_update(Relation relation, ItemPointer otid,
      |             ^~~~~~~~~~~~~~~~~~
concurrent.c:408:51: error: too few arguments to function 'ExecInsertIndexTuples'
  408 |                                         recheck = ExecInsertIndexTuples(
      |                                                   ^~~~~~~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/executor/executor.h:630:14: note: declared here
  630 | extern List *ExecInsertIndexTuples(ResultRelInfo *resultRelInfo,
      |              ^~~~~~~~~~~~~~~~~~~~~
make[1]: *** [<builtin>: concurrent.o] Error 1
make[1]: *** Waiting for unfinished jobs....
concurrent.c:311:15: error: too few arguments to function call, expected 8, have 7
                                                                                        NIL);
                                                                                           ^
/usr/pgsql-16/include/server/executor/executor.h:630:14: note: 'ExecInsertIndexTuples' declared here
extern List *ExecInsertIndexTuples(ResultRelInfo *resultRelInfo,
             ^
concurrent.c:392:44: error: too few arguments to function call, expected 4, have 3
                                simple_heap_update(relation, &ctid, tup);
                                ~~~~~~~~~~~~~~~~~~                     ^
/usr/pgsql-16/include/server/access/heapam.h:278:13: note: 'simple_heap_update' declared here
extern void simple_heap_update(Relation relation, ItemPointer otid,
            ^
concurrent.c:422:17: error: too few arguments to function call, expected 8, have 7
                                                                                                        NIL);
                                                                                                           ^
/usr/pgsql-16/include/server/executor/executor.h:630:14: note: 'ExecInsertIndexTuples' declared here
extern List *ExecInsertIndexTuples(ResultRelInfo *resultRelInfo,
             ^
3 errors generated.
make[1]: *** [/usr/pgsql-16/lib/pgxs/src/makefiles/../../src/Makefile.global:1080: concurrent.bc] Error 1
pg_squeeze.c:2528:15: error: call to undeclared function 'pg_tablespace_aclcheck'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
                aclresult = pg_tablespace_aclcheck(tablespace, relowner, ACL_CREATE);
                            ^
1 error generated.

**Comment by df7cb** [comment]
There seem to be problems on 32-bit, investigating.

**Comment by ahouska** [comment]
Fixed in 1.6.1: https://github.com/cybertec-postgresql/pg_squeeze/releases/tag/REL1_6_1

**Comment by devrimgunduz** [comment]
Thanks a lot!
